src/main/java/duke/command/CommandParser.java

```diff
@@ -0,0 +1,45 @@
+package duke.command;
+
+import duke.DukeException;
+
+/**
+ * Represents a command parser that can parse user input into a command.
+ */
+public class CommandParser {
+
+    /**
+     * Parses the given input into a command.
+     *
+     * @param input The input to be parsed.
+     * @return The parsed command.
+     * @throws DukeException If the input is invalid.
+     */
+    public Command parse(String input) throws DukeException {
+        String[] tokens = input.split(" ", 2);
+        String name = tokens[0];
+        String body = tokens.length > 1 ? tokens[1] : "";
+        switch (name) {
```

Comment:
I really like how neat this part of the code is, based on all the different cases and the calling of the constructor for the respective command classes 👍